src/object/cli_shard.c

```diff
@@ -1895,6 +1897,17 @@ obj_shard_query_recx_post(struct obj_query_key_cb_args *cb_args, uint32_t shard,
 		tmp_recx->rx_idx = rx_idx;
 		tmp_recx->rx_nr = stripe_rec_nr *
 				     (reply_recx->rx_nr / cell_rec_nr);
+
+		if (DAOS_RECX_END(*punched_recx) < DAOS_RECX_END(*tmp_recx)) {
+			tmp_recx->rx_idx = DAOS_RECX_END(*punched_recx);
+			tmp_recx->rx_nr = DAOS_RECX_END(*tmp_recx) - tmp_recx->rx_idx;
+		} else if (punched_recx->rx_idx > tmp_recx->rx_idx) {
+			tmp_recx->rx_nr = min(punched_recx->rx_idx, DAOS_RECX_END(*tmp_recx)) -
+					  tmp_recx->rx_idx;
+		} else {
+			tmp_recx->rx_nr = 0;
+			tmp_recx->rx_idx = 0;
```

Comment:
this case seems not fully correct.
For example, EC 2+1, cell_size 1M,
user firstly partial write [1M ~ 2M], then full-stripe write [4M ~ 6M], and punch [4M ~ 6M]
then query the MIN recx, it should get [1M ~ 2M], but after here's reset to 0, the final result seem swill be [0, 0], seems incorrect.

Reply:
hmm, in this case, recx[0] will be [1M - 2M], recx[1] will be[0, 0], so recx0 will be chosen? no?

Reply:
true for GET_MAX case, but if without GET_MAX, then should get the minimal [1M - 2M], but the code here will cause [0, 0] replied, no?